Google.Api.Generator/Generation/SnippetCodeGenerator.cs

```diff
@@ -35,102 +35,137 @@ namespace Google.Api.Generator.Generation
                             }
                         }
                         break;
                     case MethodDetails.Paginated _:
-                        yield return methodDef.SyncPaginatedRequestMethod;
-                        yield return methodDef.AsyncPaginatedRequestMethod;
+                        yield return methodDef.SyncPaginatedRequestSnippet;
+                        yield return methodDef.AsyncPaginatedRequestSnippet;
                         foreach (var signature in methodDef.Signatures)
                         {
-                            yield return signature.SyncPaginatedMethod;
-                            yield return signature.AsyncPaginatedMethod;
+                            yield return signature.SyncPaginatedSnippet;
+                            yield return signature.AsyncPaginatedSnippet;
                             foreach (var resourceName in signature.ResourceNames)
                             {
-                                yield return resourceName.SyncPaginatedMethod;
-                                yield return resourceName.AsyncPaginatedMethod;
+                                yield return resourceName.SyncPaginatedSnippet;
+                                yield return resourceName.AsyncPaginatedSnippet;
                             }
                         }
                         break;
                     case MethodDetails.ServerStreaming _:
-                        yield return methodDef.ServerStreamingRequestMethod;
+                        yield return methodDef.ServerStreamingRequestSnippet;
                         foreach (var signature in methodDef.Signatures)
                         {
-                            yield return signature.ServerStreamingMethod;
+                            yield return signature.ServerStreamingSnippet;
                             foreach (var resourceName in signature.ResourceNames)
                             {
-                                yield return resourceName.ServerStreamingMethod;
+                                yield return resourceName.ServerStreamingSnippet;
                             }
                         }
                         break;
                     case MethodDetails.BidiStreaming _:
-                        yield return methodDef.BidiStreamingMethod;
+                        yield return methodDef.BidiStreamingSnippet;
                         break;
                 }
             }
         }
 
+        private SnippetCodeGenerator(SnippetDef snippetDef) => 
+            Snippet = snippetDef;
+
+        private SnippetDef Snippet { get; }
+        public string SnippetName => Snippet.SnippetName;
+
+        public CompilationUnitSyntax Generate(SourceFileContext ctx) =>
+            ctx.CreateCompilationUnit(Snippet.StandaloneSnippet(ctx));
+
+        private class SnippetDef
+        {
+            public string SnippetName { get; }
```

Comment:
What is the "name" of a snippet? I can think of a bunch of things this could refer to - a comment with more details and an example would be really helpful.